pkg/kubectl/resource/mapper.go

```diff
@@ -62,9 +62,12 @@ func (m *Mapper) InfoForData(data []byte, source string) (*Info, error) {
 	if err != nil {
 		return nil, fmt.Errorf("unable to load %q: %v", source, err)
 	}
-	client, err := m.ClientForMapping(mapping)
-	if err != nil {
-		return nil, fmt.Errorf("unable to connect to a server to handle %q: %v", mapping.Resource, err)
+	var client RESTClient
+	if !local {
+		client, err = m.ClientForMapping(mapping)
```

Comment:
Why can't I get a RESTClient in these cases?  Is it because of the API version negotiation calls also noted here: https://github.com/kubernetes/kubernetes/issues/13794 ?